qa/tasks/ec_parity_consistency.py

```diff
@@ -0,0 +1,742 @@
+
+
+def shell(ctx: Any, cluster_name: str, remote: Any,
+          args: List[str], name: str = "", **kwargs: Any):
+    """
+    Interface for running commands on cephadm clusters
+    """
+    extra_args = []
+    if name:
+        extra_args = ['-n', name]
+    return remote.run(
+        args=[
+            'sudo',
+            ctx.cephadm,
+            '--image', ctx.ceph[cluster_name].image,
+            'shell',
+        ] + extra_args + [
+            '--fsid', ctx.ceph[cluster_name].fsid,
+            '--',
+        ] + args,
+        **kwargs
+    )
+
+
+def get_tmp_directory():
+    """
+    Returns a temporary directory name that will be used to store shard data
+    Includes the PID so different instances can be run in parallel
+    """
+    tmpdir = '/var/tmp/consistency-check-' + str(os.getpid()) + '/'
+    return tmpdir
+
+
+def handle_mismatch(assert_on_mismatch: bool):
+    """
+    Raise a RunTimeError if assert_on_mismatch is set,
+    otherwise just log an error.
+    """
+    err = "Shard mismatch detected."
+    if assert_on_mismatch:
+        raise RuntimeError(err)
+    log.error(err)
+
+
+def load_objects_to_check(objects_to_check: str) -> List[str]:
+    """
+    Attempt to parse the object_list config option
+    takes a stringified JSON list as an argument
+    """
+    object_list = []
+    if objects_to_check is None:
+        return object_list
+    try:
+        object_list = json.loads(objects_to_check)
+    except ValueError:
+        log.error("Failed to parse object_list config option,"
+                  "input should be a JSON formatted list")
+    return object_list
+
+
+def revive_osds(manager: ceph_manager.CephManager, osds: List[Dict[str, Any]]):
+    """
+    Revive any OSDs that were killed during the task and
+    clean up any temporary files (temp files both locally and on remote).
+    Optionally retain files for debug if specified by the config
+    """
+    for osd in osds:
+        osd_id = osd["osd"]
+        manager.revive_osd(osd_id, skip_admin_check=False)
+        manager.mark_in_osd(osd_id)
+        manager.wait_till_osd_is_up(osd_id)
+
+
+def clean_up_test_files(remote: Any = None):
+    """
+    Clean any test files that were created
+    both locally and on a remote if specified
+    """
+    tmp_dir = get_tmp_directory()
+    local_dir_exists = os.path.isdir(tmp_dir)
+    if local_dir_exists:
+        shutil.rmtree(tmp_dir)
+    if remote:
+        remote.run(args=["rm", "-rf", tmp_dir])
+
+
+def print_summary(consistent: List[str],
+                  inconsistent: List[str],
+                  skipped: List[str]):
+    """
+    Print a summary including counts of objects checked
+    and a JSON-formatted lists of consistent and inconsistent objects.
+    """
+    log.info("Consistent objects counted: %i", len(consistent))
+    log.info("Inconsistent objects counted %i", len(inconsistent))
+    log.info("Objects skipped: %i", len(skipped) + len(skipped))
```

Comment:
```suggestion
    log.info("Objects skipped: %i", len(skipped) + len(skipped))
```
Unclear on why we're adding `skipped` with `skipped`?